src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject.cs

```diff
@@ -591,6 +597,28 @@ private void OnImportChanged(object sender, EventArgs e)
         {
             VisualStudioMetadataReference reference = (VisualStudioMetadataReference)sender;
 
+            CancellationTokenSource delayTaskCancellationTokenSource;
+            if (_changedReferencesPendingUpdate.TryGetValue(reference, out delayTaskCancellationTokenSource))
```

Comment:
Which thread is this being raised on? Does it need locking? My gut would say it's being called on a thread pool thread, which means we already had synchronization issues. :-)